### DISCUSSION TITLE
is sponge schematic supported?

### DISCUSSION THREAD

**Discussion by kasterialgit** [maruohon/litematica#694] [Q&A] [answered]
is sponge schematic format supported in all version of litematica?

**Comment by maruohon** [comment, marked as answer]
No, not yet.

Currently the 1.12.2 LiteLoader version has proper "full" support for that format, meaning both reading and writing, and treating it "properly" internally.

The 1.16+ (or was it 1.17+ ?) versions have a temporary quick hack support for reading the Sponge format, and basically immediately converting it to the Litematica format when loading.

The proper full support from 1.12.2 will come to the later MC versions whenever I'm ready to start porting the 1.12.2 code. There is still lots of code clean-up and some rewrites planned, not to mention a long todo list of new features and user-facing improvements. It's likely that it will first come to 1.19 or 1.20 in the form of "early test ports", and the full port to all older MC versions will happen later when I finish more/all(?) of my todo list for the mod first in the main development branch in 1.12.2.

**Reply by kasterialgit** [reply]
Ok sounds good.
